src/Chess.cpp

```diff
@@ -0,0 +1,257 @@
+		m_board[20][0] = 200; m_board[20][20] = 188;
+
+		for (size_t i = 1; i < 20; ++i)
+		{
+			m_board[0][i] = 205;
+			m_board[20][i] = 205;
+			m_board[i][0] = 186;
+			m_board[i][20] = 186;
+		}
+	} 
+
+	// set in side frame 
+	{
+		m_board[2][2] = 218;  m_board[2][18] = 191;
+		m_board[18][2] = 192; m_board[18][18] = 217;
+
+		for (size_t i = 4; i < 17; i += 2)
+		{
+			m_board[2][i] = 194;
+			m_board[18][i] = 193;
+			m_board[i][2] = 195;
+			m_board[i][18] = 180;
+		}
+
+		for (size_t i = 2; i < 19; i += 2)
+			for (size_t j = 3; j < 19; j += 2)
+				m_board[i][j] = 196;
+		
+		for (size_t i = 3; i < 18; i += 2)
+			for (size_t j = 2; j < 19; j += 2)
+				m_board[i][j] = 179;
+
+		for (size_t i = 4; i < 17; i += 2)
+			for (size_t j = 4; j < 17; j += 2)
+				m_board[i][j] = 197;
+
+		for (size_t i = 4; i < 17; i += 2)
+			m_board[2][i] = 194;
+		for (size_t i = 4; i < 17; i += 2)
+			m_board[18][i] = 193;
+		for (size_t i = 4; i < 17; i += 2)
+			m_board[i][2] = 195;
+		for (size_t i = 4; i < 17; i += 2)
+			m_board[i][18] = 180;
+	}
+
+	// set nums and letters 
+	{
+		// nums 
+		for (size_t i = 3, t = 0; i < 19; i += 2, ++t)
+			m_board[1][i] = m_board[19][i] = ('1' + t);
+		// letters 
+		for (size_t i = 3, t = 0; i < 19; i += 2, ++t)
+			m_board[i][1] = m_board[i][19] = ('A' + t);
+	}
+}
+// put the pieces from the boardString 
+void Chess::setPieces()
+{
+	for (size_t row = 0, t = 0; row < 8; ++row)
+		for (size_t col = 0; col < 8; ++col, ++t)
+			m_board[(3 + (row * 2))][(3 + (col * 2))] = ((m_boardString[t] == '#') ? 32 : m_boardString[t]);
+}
+// print the only the board to screen 
+void Chess::show() const 
+{
+	for (size_t row = 0; row < _SIZE; ++row)
+	{
+		for (size_t col = 0; col < _SIZE; ++col)
+			cout << m_board[row][col];
+		cout << endl;
+	}
+}
+// clear screen and print the board and the relevant msg 
+void Chess::displayBoard() const
+{
+	clear();
+	show();
+	cout << m_msg<< m_errorMsg;
+	
+}
+// print the who is turn before getting input 
+void Chess::showAskInput() const 
+{
+	if (m_turn)
+		cout << "Player 1 (White - Capital letters) >> ";
+	else
+		cout << "Player 2 (Black - Small letters)   >> ";
+}
+// check if the source and dest are the same 
+bool Chess::isSame() const 
+{
+	return ((m_input[0] == m_input[2]) && (m_input[1] == m_input[3]));
+} 
+// check if the input is lockations at board
+bool Chess::isValid() const
+{
+	return ((('A' <= m_input[0]) && (m_input[0] <= 'H')) || (('a' <= m_input[0]) && (m_input[0] <= 'h')) &&
+		(('1' <= m_input[1]) && (m_input[1] <= '8')) &&
+		(('A' <= m_input[2]) && (m_input[2] <= 'H')) || (('a' <= m_input[2]) && (m_input[2] <= 'h')) &&
+		(('1' <= m_input[3]) && (m_input[3] <= '8')));
+}
+	
+// check if the input is exit or quit  
+bool Chess::isExit() const 
+{
+	return ((m_input == "exit") || (m_input == "quit") || (m_input == "EXIT") || (m_input == "QUIT"));
+}
+// execute the movement on board 
+void Chess::excute()
+{
+	int row = (m_input[0] - 'a');

```

Comment:
Use descriptive variable names: Instead of using generic names like row, col consider using more descriptive names that convey their purpose and meaning. For example, startRow could be sourceRow, endRow could be destinationRow, and so on. This improves code readability and makes it easier to understand the logic.